sources/zafira-services/src/main/java/com/qaprosoft/zafira/service/integration/core/IntegrationTenancyStorage.java

```diff
@@ -104,10 +105,18 @@ public void initDb() {
 
     @RabbitListener(queues = "#{settingsQueue.name}")
     public void process(Message message) {
-        ReinitEventMessage rm = new Gson().fromJson(new String(message.getBody()), ReinitEventMessage.class);
-        if (!eventPushService.isSettingQueueConsumer(message)) {
-            Integration integration = integrationService.retrieveById(rm.getIntegrationId());
-            integrationInitializer.initIntegration(integration, rm.getTenancy());
+        try {
+            ReinitEventMessage rm = new Gson().fromJson(new String(message.getBody()), ReinitEventMessage.class);
+            if (!eventPushService.isSettingQueueConsumer(message)) {
+                TenancyContext.setTenantName(rm.getTenancy());
+
+                Integration integration = integrationService.retrieveById(rm.getIntegrationId());
+                integrationInitializer.initIntegration(integration, rm.getTenancy());
+
+                TenancyContext.setTenantName(null);
```

Comment:
What the problem we usually fix setting tenant name as null? Noticed it in a coupe other places, once even as causing bug.

Reply:
@tsvrko no problems, just be sure that thread maybe stored in a rabbit pool does not have a tenant state (additional check)